### PR TITLE
#1023@minor: Adds support for dispatching a click event on the contro…

### PR DESCRIPTION
**Pull request by capricorn86**
…l element when clicking on a label.

### CODE REVIEW

File: packages/happy-dom/src/nodes/html-label-element/HTMLLabelElement.ts [hunk 1]
```diff
@@ -66,4 +70,23 @@ export default class HTMLLabelElement extends HTMLElement implements IHTMLLabelE
...
+	public override dispatchEvent(event: Event): boolean {
+		const returnValue = super.dispatchEvent(event);
+
+		if (
+			event.type === 'click' &&
+			(event.eventPhase === EventPhaseEnum.atTarget || event.eventPhase === EventPhaseEnum.bubbling)
+		) {
+			const control = this.control;
+			if (control && event.target !== control) {
+				control.dispatchEvent(new PointerEvent('click', { bubbles: true, cancelable: true }));
```
Comment (timkindberg) [inline packages/happy-dom/src/nodes/html-label-element/HTMLLabelElement.ts:86]: Is this what happens in native browsers as well? It dispatches a click event on the input as well?

Comment (timkindberg) [reply packages/happy-dom/src/nodes/html-label-element/HTMLLabelElement.ts:86]: Yup ok, looks like it is the correct behavior! https://codesandbox.io/s/quizzical-cerf-mp2xtk?file=/src/App.js

https://github.com/capricorn86/happy-dom/assets/168046/3ad2c2fc-98db-4f32-b655-5b35f44239f3

Comment (capricorn86) [reply packages/happy-dom/src/nodes/html-label-element/HTMLLabelElement.ts:86]: Took sometime to figure it out, but it should be correct now :slightly_smiling_face:

Comment (capricorn86) [reply packages/happy-dom/src/nodes/html-label-element/HTMLLabelElement.ts:86]: The label will have two click event dispatched on it if it is a parent to the input, as the event bubbles up to it.

Comment (capricorn86) [reply packages/happy-dom/src/nodes/html-label-element/HTMLLabelElement.ts:86]: Yes sure, it supports that as well.

File: packages/happy-dom/test/nodes/html-label-element/HTMLLabelElement.test.ts [hunk 2]
```diff
@@ -76,4 +78,30 @@ describe('HTMLLabelElement', () => {
 			expect(element.form === form).toBe(true);
 		});
 	});
+
+	describe('dispatchEvent()', () => {
+		it('Dispatches a click event on the control element if it exists.', () => {
```
[the remark this answers is not in the document]
Comment (artursvonda) [reply packages/happy-dom/test/nodes/html-label-element/HTMLLabelElement.test.ts:83]: Nevermind me, understood what happened here. `span` is nested inside the label. Sorry for silly question.

[the remark this answers is not in the document]
Comment (capricorn86) [reply packages/happy-dom/test/nodes/html-label-element/HTMLLabelElement.test.ts:83]: @artursvonda what is it that broke? Can you file a bug for it?

[the remark this answers is not in the document]
Comment (artursvonda) [reply packages/happy-dom/test/nodes/html-label-element/HTMLLabelElement.test.ts:83]: Looks like issue might be with `@testing-library/user-event` dispatching click event twice on their own causing checkbox to be toggled twice. Still investigating and once I have repro, will creating issue on both repos.

[the remark this answers is not in the document]
Comment (capricorn86) [reply packages/happy-dom/test/nodes/html-label-element/HTMLLabelElement.test.ts:83]: Thank you!

**Pull request merged**